sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md

```diff
@@ -0,0 +1,389 @@
+
+```csharp
+var client = CreateClient();
+var producer = client.CreateProducer();
+
+using (var eventBatch = CreateAndFillBatch(producer))
+{
+    var maximumSizeInBytes = eventBatch.MaximumSize;
+    var batchSizeInBytes = eventBatch.CurrentSize;
+}
+```
+
+#### Publish a batch
+
+```csharp
+var client = CreateClient();
+var producer = client.CreateProducer();
+
+using (var eventBatch = CreateAndFillBatch(producer))
+{
+    // Options when sending an EventBatch are allowed only
+    // on the batch itself, they may not be also provided during
+    // the Send.
+    
+    await producer.Send(eventBatch);
+}
+```
+
+### Examples: Consuming Events from a Partition
+
+#### Subscribe to events as they are available, waiting as needed 
+
+```csharp
+var client = CreateClient();
+var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroupName, "fcbac12-43cda", EventPosition.Earliest);
+
+await foreach(var eventData in consumer.SubscribeToPartition(someCancellationToken))
+{
+    ProcessEvent(eventData);
+}
+```
+
+#### Subscribe to events as they are available, using a maximum wait time to return control
+
+```csharp
+var client = CreateClient();
+var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroupName, "fcbac12-43cda", EventPosition.Earliest);
+var maximumWaitTime = TimeSpan.FromSeconds(2);
+
+await foreach(var eventData in consumer.SubscribeToPartition(maximumWaitTime, someCancellationToken))
+{
+    // If the maximum wait time elapsed before an event was 
+    // available, it will be null.  This returns control and allows
+    // for breaking out of the loop.
+    
+    if (eventData != null)
+    {
+        ProcessEvent(eventData);
+    }
+}
+```
+
+### Packages
+
+#### `Azure.Messaging.EventHubs`
+
+The main client library package, containing the core components for interacting with the Azure Event Hubs service.
+
+### Namespaces
+
+#### `Azure.Messaging.EventHubs`
+
+The top-level container for the types in the client library intended to be used by Event Hubs client library users. It is intended that types most frequently used by users for basic operations belong to this namespace to ensure ease of discovery.
+
+##### _Example Types:_
+ 
+- `RetryOptions`
+- `EventHubsRetryPolicy`
+- `BatchOptions`
+- `EventDataBatch`
+
+#### `Azure.Messaging.EventHubs.Errors`
+
+The location for exceptions and error-related information and operations. Many of these types are surfaced as information for consumers in response to conditions encountered during a basic operation. It is not intended that consumers need to create these types directly.
+
+##### _Example Types:_
+ 
+- `OperationCancelledException`
+- `TimeoutException`
+- `MessageSizeExceededException`
+- `QuotaExceededException`
+
+#### `Azure.Messaging.EventHubs.Core`
+
+The location for internal types used by the Event Hubs library to facilitate operations; these constructs are not intended to be consumed externally. 
+
+##### _Example Types:_

```

Comment:
I think you were using sentence casing for your headers, so would this be "Example types:"?

Reply:
Yeah, a bit inconsistent there.   Thanks!